scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1759,7 +1759,15 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         }
         if (looksLikeLambda) {
           next()
-          t = atPos(t, auto)(Term.Function(convertToParams(t), if (location != InBlock) expr() else block()))
+          t = atPos(t, auto)({
+
+            val term = (if (location != InBlock) expr() else block()) match {
+              // Don't synthesize a block in a case expression if there is only one expression
```

Comment:
cc/ @xeno-by I'm not really sure if this is the right solution, but it seems to do the right thing. I'm sure you know this better than me.